src/allmydata/scripts/create_node.py

```diff
@@ -218,8 +263,27 @@ def parseArgs(self, basedir=None):
         validate_i2p_options(self)
 
 
-@defer.inlineCallbacks
-def write_node_config(c, config):
+def merge_config(
+        left: Optional[ListenerConfig],
+        right: Optional[ListenerConfig],
+) -> Optional[ListenerConfig]:
+    """
+    Merge two listener configurations into one configuration representing
+    both of them.
+
+    If either is ``None`` then the result is ``None``.  This supports the
+    "disable listeners" functionality.
+    """
+    if left is None or right is None:
+        return None
+    return ListenerConfig(
+        list(left.tub_ports) + list(right.tub_ports),
+        list(left.tub_locations) + list(right.tub_locations),
+        dict(list(left.node_config.items()) + list(right.node_config.items())),
```

Comment:
What happens if they have keys in common?

Reply:
Bad stuff!  It won't happen given the configs currently generated ... but I changed it to raise an exception if it does, so it isn't a silent failure if it ever comes up.